types/office-js-preview/index.d.ts

```diff
@@ -525,14 +525,55 @@ declare namespace Office {
         value: T;
     }
     /**
-     * Used to associate an action name to a function.
+     * Manages actions and keyboard shortcuts.
      */
     interface Actions {
         /**
-         * Function to associate a name with the action function.
+         * Associates the ID of an action with a function.
+         * 
+         * @param actionId The ID of an action that is defined in an extended manifest.
+         * @param actionFunction The function that is run when the action is invoked. 
          */
-        associate: (actionName: string, action: (arg?: any) => void) => void;
-    }
+        associate: (actionId: string, actionFunction: (arg?: any) => void) => void;
+        /**
+         * Replaces existing add-in shortcuts with custom shortcuts for the user.
+         *
+         * @remarks
+         *
+         * **Requirement set**: KeyboardShortcuts 1.1
+         * @beta
+         * 
+         * @param shortcuts An object of custom shortcuts with keys being the IDs of the actions (as defined in an extended manifest) and values being the shortcut combinations; for example, {"SetItalic": "Ctrl+1", "SetBold": "Ctrl+2"};
+         * To learn how to specify a valid action ID and a key combination, see {@link https://docs.microsoft.com/office/dev/add-ins/design/keyboard-shortcuts | Add custom keyboard shortcuts to your Office Add-ins}. (Note tht a key combination can be `null`, in which case, the action keeps the key combination specified in the JSON file.)
+         * @returns A promise that resolves when every custom shortcut assignment in `shortcuts` has been registered. Even if there is a conflict with existing shortcuts, the customized shortcut will be registered;
+         * otherwise, the promise will be rejected with error code and error message; an error code "InvalidOperation" is returned if any action ID in `shortcuts` does not exist, or if shortcut combination is invalid.
+         */
+        replaceShortcuts(shortcuts: {[actionId:string]:string}): Promise<void>;   
+        /**
+         * Gets the existing shortcuts for the add-in. The set always includes (1) the shortcuts defined in the add-in's extended manifest for keyboard shortcuts and (2) the current user's custom shortcuts if those exist.
+         * The shortcut can be `null` if it conflicts with the shortcut of another add-in or with the Office application. specifically, it would be `null` if, when prompted to choose which shortcut to use, the user didn't choose the action of the current add-in. For more information about conflicts with shortcuts, see  {@link https://docs.microsoft.com/office/dev/add-ins/design/keyboard-shortcuts#avoid-key-combinations-in-use-by-other-add-ins | Avoid key combinations in use by other add-ins}.
```

Comment:
```suggestion
         * The shortcut can be `null` if it conflicts with the shortcut of another add-in or with the Office application. Specifically, it would be `null` if, when prompted to choose which shortcut to use, the user didn't choose the action of the current add-in. For more information about conflicts with shortcuts, see {@link https://docs.microsoft.com/office/dev/add-ins/design/keyboard-shortcuts#avoid-key-combinations-in-use-by-other-add-ins | Avoid key combinations in use by other add-ins}.
```